spec.html

```diff
@@ -13403,21 +13564,23 @@ <h1>
       <h1>
         DefineMethodProperty (
           _homeObject_: an Object,
-          _key_: a property key or Private Name,
-          _closure_: a function object,
+          _methodDefinition_: a ClassElementDefinition Record,
           _enumerable_: a Boolean,
-        ): a PrivateElement or ~unused~
+        ): either a normal completion containing ~unused~, or an abrupt completion
       </h1>
       <dl class="header">
       </dl>
       <emu-alg>
         1. Assert: _homeObject_ is an ordinary, extensible object with no non-configurable properties.
-        1. If _key_ is a Private Name, then
-          1. Return PrivateElement { [[Key]]: _key_, [[Kind]]: ~method~, [[Value]]: _closure_ }.
-        1. Else,
-          1. Let _desc_ be the PropertyDescriptor { [[Value]]: _closure_, [[Writable]]: *true*, [[Enumerable]]: _enumerable_, [[Configurable]]: *true* }.
-          1. Perform ! DefinePropertyOrThrow(_homeObject_, _key_, _desc_).
-          1. Return ~unused~.
+        1. Assert: _methodDefinition_.[[Kind]] is ~method~, ~getter~, ~setter~, or ~accessor~.
+        1. Let _key_ be _methodDefinition_.[[Key]].
+        1. If _key_ is not a Private Name, then
+          1. Let _desc_ be the PropertyDescriptor { [[Writable]]: *true*, [[Enumerable]]: _enumerable_, [[Configurable]]: *true* }.
+          1. If _methodDefinition_.[[Kind]] is ~getter~ or ~accessor~, set _desc_.[[Get]] to _methodDefinition_.[[Get]].
+          1. If _methodDefinition_.[[Kind]] is ~setter~ or ~accessor~, set _desc_.[[Set]] to _methodDefinition_.[[Set]].
+          1. If _methodDefinition_.[[Kind]] is ~method~, set _desc_.[[Value]] to _methodDefinition_.[[Value]].
```

Comment:
[[Writable]] is invalid on an accessor descriptor.
```suggestion
          1. Let _desc_ be the PropertyDescriptor { [[Enumerable]]: _enumerable_, [[Configurable]]: *true* }.
          1. If _methodDefinition_.[[Kind]] is ~getter~ or ~accessor~, set _desc_.[[Get]] to _methodDefinition_.[[Get]].
          1. If _methodDefinition_.[[Kind]] is ~setter~ or ~accessor~, set _desc_.[[Set]] to _methodDefinition_.[[Set]].
          1. If _methodDefinition_.[[Kind]] is ~method~, set _desc_.[[Value]] to _methodDefinition_.[[Value]] and _desc_.[[Writable]] to *true*.
```